### ISSUE TITLE
Factorio credentials are not wiped if in instance config

### ISSUE THREAD

**Issue by Cooldude2606** [clusterio/clusterio#724]
TL:DR if the factorio credentials are included in the instance config then they are always included in the server settings json even after "being wiped".

This is caused by the config being seen as an "override" and the entries are not filtered to exclude the username, password, or token. To fix this would require a conditional filter being applied based on if credentials should be included. See `resolveServerSettings`

**Comment by Hornwitser** [comment]
This was done by design when I implemented the logic. The reasoning behind this is that the host stores a complete copy of the instance config in `instance.json` which is located right next to the `server-settings.json` file. The credentials stored in the instance config ends up in both of these files, so removing them from one didn't make much sense.

**Comment by Cooldude2606** [comment]
Does this still hold if the config files are no longer world readable? (as is planned at some point)

**Comment by Hornwitser** [comment]
Yes, I would assume `server-settings.json` would no longer be world readable just like the instance config.

**Comment by Cooldude2606** [comment]
Ah, well if the file is always going to have the same permissions as the config; then I understand the reasoning and can close this.
